### DISCUSSION TITLE
How to import a published package into an app?

### DISCUSSION THREAD

**Discussion by oriyan-ovadia-healthy** [vercel/turborepo#1164] [Help]
We have a monorepo with published packages. We also import these packages directly in the **apps** folder.

```
/apps
  /app1
  /app2
/packages
  /ui
```

### What's the problem?

Our imports _within apps_ doesn't work without writing `/src`.

```js
// Doesn't work
import { Button } from '@my-npm/ui';

// Does work
import { Button } from '@my-npm/ui/src';
```

When we import the package into a *different repo*, it works as expected.

```js
import { Button } from '@my-npm/ui';
```

### How does a package look like?

```
/ui
  /src
    /Button
      index.js
    index.js
  package.json
```

_package.json -_
```json
"main": "lib/index.js",
```

### How did we decide to fix it?

By adding an `index.js` file at the package's root, that exports `src`.

```js
// ui/index.js
export * from './src';
```

But what we'd like to know is, can we achieve it without our fix?

**Comment by lodmfjord** [comment]
Change "main" to ui/src/index.js ?

This is not something related to TurboRepo though, I would imagine.

**Comment by anthonyshew** [comment]
Published packages do take a little more configuration to make work in-repo and for publishing, as you've noted. You've mentioned the published package works which is great!

Unfortunately, I can't say for sure what exact configuration will work best for you as we would need more info. A reproduction repo would be great but, given that this Discussion is nearly 1.5 years old, this may be stale.

I'll close it for now and re-open if there is still interest. 😄
